product_secondary_unit/models/product_secondary_unit_mixin.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_round
+
+
+class StockSecondaryUnitMixin(models.AbstractModel):
+    """
+    Mixin model that allow to compute a field from a se secondary unit helper
+    Example to extend any model in which you want to compute quantities based
+    on secondary units. You must add class attribute _secondary_unit_fields
+
+    _secondary_unit_fields = {
+        "qty_field": "product_uom_qty",
+        "uom_field": "product_uom"
+    }
+    """
+
+    _name = "product.secondary.unit.mixin"
+    _description = "Product Secondary Unit Mixin"
+    _secondary_unit_fields = {}
+
+    secondary_uom_qty = fields.Float(
+        string="Secondary Qty",
+        digits="Product Unit of Measure",
+        store=True,
+        readonly=False,
+        compute="_compute_secondary_uom_qty",
+    )
+    secondary_uom_id = fields.Many2one(
+        comodel_name="product.secondary.unit", string="Second unit"
+    )
+
+    def _get_uom_line(self):
+        return self[self._secondary_unit_fields["uom_field"]]
+
+    def _get_factor_line(self):
+        return self.secondary_uom_id.factor * self._get_uom_line().factor
+
+    def _get_quantity_from_line(self):
+        return self[self._secondary_unit_fields["qty_field"]]
+
+    @api.model
+    def _get_secondary_uom_qty_depends(self):
+        if not self._secondary_unit_fields:
+            return []
+        return [self._secondary_unit_fields["qty_field"]]
+
+    @api.depends(lambda x: x._get_secondary_uom_qty_depends())
+    def _compute_secondary_uom_qty(self):
+        for line in self:
+            if not line.secondary_uom_id:
+                continue
+            factor = line._get_factor_line()
+            qty_line = line._get_quantity_from_line()
+            qty = float_round(
+                qty_line / (factor or 1.0),
+                precision_rounding=line.secondary_uom_id.uom_id.rounding,
+            )
+            line.secondary_uom_qty = qty
+
+    def _compute_target_field_qty(self):
+        """
+        Set the target qty field defined in model
+        """
+        for rec in self:
+            if not rec.secondary_uom_id:
+                continue
+            factor = rec._get_factor_line()
+            qty = float_round(
+                rec.secondary_uom_qty * factor,
+                precision_rounding=rec._get_uom_line().rounding,
+            )
+            rec[rec._secondary_unit_fields["qty_field"]] = qty
+
+    def _onchange_product_uom_for_secondary(self):
+        """
+        Helper method to be called from onchange method of uom field in
+        target model
+        """
```

Comment:
Please use PEP257 for docstrings:
```suggestion
        """Helper method to be called from onchange method of uom field in
        target model.
        """
```